system/security/shared/caching.hpp

```diff
@@ -151,16 +151,25 @@ class CPermissionsCache
 
     virtual ~CPermissionsCache();
 
-    static CPermissionsCache* queryInstance()
+    //Returns a shared cache of a given class type.
+    //Call this method with a unique class string ("LDAP", "MyOtherSecMgr")
+    //to create a cache shared amongst security managers of the same class
+    static CPermissionsCache* queryInstance(const char * _secMgrClass)
     {
+        const char * secMgrClass = (_secMgrClass != nullptr  &&  *_secMgrClass) ? _secMgrClass : "genericSecMgrClass";
+        typedef map<string, CPermissionsCache*> MapCache;
+        static MapCache m_mapCache;
```

Comment:
See previous comments about local static variables...

Also, when are these objects released? If the answer is "Never" then it will hinder checking the system for memory leaks... (I think that is also an issue with the previous code).